base/abstractarray.jl

```diff
@@ -1472,17 +1471,17 @@ function _typed_hcat(::Type{T}, A::AbstractVecOrTuple{AbstractVecOrMat}) where T
         nd = ndims(Aj)
         ncols += (nd==2 ? size(Aj,2) : 1)
     end
-    B = similar(A[1], T, nrows, ncols)
-    pos = 1
+    B = similar(first(A), T, nrows, ncols)
```

Comment:
How difficult would it be to adapt this to have `B = similar(first(A), T, axes(first(A),1), axes(A,1))`, so that all the offsets get propagated?

I guess that's only the case of a vector of vectors; when some elements are matrices you'd have to give up on the 2nd axis.

Reply:
As explained in https://github.com/JuliaArrays/OffsetArrays.jl/issues/63#issuecomment-689806145, a meaningful offset propagation is only possible for a very restricted conditions. This restriction doesn't make `cat`s of OffsetArrays indeed useful.

Reply:
Oh I didn't see the issue, let me look there first.